### PR TITLE
XEP 0107 - User Mood

### PR DESCRIPTION
**Pull request by DebXWoody**
Implementation of XEP 0107 - User Mood

### CODE REVIEW

**Comment by DebXWoody** [comment]
```
        if (args[2]) {
            publish_user_mood(args[1], args[2]);
        } else {
            publish_user_mood(args[1], args[1]);
        }
```
If the user added additional text `/mood set happy "I like profanity!"`, we will use this as text.

`13:20:11 - Mood from user@domain.tld happy (I like profanity!)`

If the user didn't add additional text `/mood set happy` we will add the mood as text.

`13:20:02 - Mood from user@domain.tld happy (happy)`

File: src/xmpp/message.c [hunk 1]
```diff
@@ -253,6 +258,8 @@ _message_handler(xmpp_conn_t* const conn, xmpp_stanza_t* const stanza, void* con
         if (msg_stanza) {
             _handle_chat(msg_stanza, FALSE, is_carbon, NULL, NULL);
         }
+    } else if (type && g_strcmp0(type, STANZA_TYPE_HEADLINE) == 0) {
```
[the remark this answers is not in the document]
Comment (jubalh) [reply src/xmpp/message.c:255]: I still don't see why we do this. If you want to tackle this late then please revert this change like it was before.

**Comment by jubalh** [comment]
@DebXWoody  Ping :)

File: src/xmpp/session.c [hunk 2]
```diff
@@ -286,6 +286,36 @@ session_get_account_name(void)
...
+                            const char* text = xmpp_stanza_get_text(t);
+                            cons_show("Mood from %s %s (%s)", from, m, text);
+                        } else {
+                            cons_show("Mood from %s %s", from, m);
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (sjaeckel) [inline src/xmpp/session.c:315]: when I saw this code I came up with strophe/libstrophe#187

Do you think this might help?

Comment (jubalh) [reply src/xmpp/session.c:315]: Yes, I like that! Thanks!

**Comment by DebXWoody** [comment]
> when I saw this code I came up with strophe/libstrophe#187
> 
> Do you think this might help?

Awesome!

File: src/xmpp/session.c (hunk 2 above)
Comment (sjaeckel) [reply src/xmpp/session.c:315]: It's merged, now we only have to wait for a new release :)

File: src/command/cmd_defs.c [hunk 3]
```diff
@@ -2688,7 +2688,7 @@ static struct cmd_t command_defs[] = {
       CMD_SYN(
               "/mood set <mood> [\"Text\"]")
       CMD_DESC(
-              "Set your mood")
+              "Set your mood. Use the tab key to switch through the available moods")
```
Comment (jubalh) [inline src/command/cmd_defs.c:2691]: Punkt vergessen ;)

**Review by jubalh — CHANGES_REQUESTED** [review]

**Review by jubalh — APPROVED** [review]
Since there was no reaction. And no review was clicked on resolved. I decided to go through the code myself. Make the last adjustments.
There is still the confusion about headline/pubsub. I made a comment in the commit. And hope @DebXWoody can take a look when he has more time.

**Comment by jubalh** [comment]
Thanks for adding this XEP @DebXWoody !

**Pull request merged**